### DISCUSSION TITLE
Is it possible to filter queries with "refetching : true" while using "useIsFetching" or "useIsMutating" hooks?

### DISCUSSION THREAD

**Discussion by mojjominion** [TanStack/query#2176] [Q&A] [answered]
Currently I'm using them as below
```
 const isFetching = useIsFetching({ fetching: true });
 const isMutating = useIsMutating({ fetching: true });
  ```
  
 What I'd like to have is queries which have cache present and are being refeched
  ```
 const isFetching = useIsFetching({ refetching: true }); 
 const isMutating = useIsMutating({ refetching: true });
  ```

**Comment by TkDodo** [comment, marked as answer]
`fetching: true` doesn't really make much sense on these hooks, because it will give you the number of hooks that are currently fetching 🙃 .

What might works is using the `predicate` function to filter down to only queries that already have data, something like:

```
useIsFetching({ predicate: (query) => query.state?.data !== undefined })
```

or you could filter down to queries that are in the `success` state:

```
useIsFetching({ predicate: (query) => query.state?.status === 'success' })
```

because only a query that is successful can be re-fetched

**Reply by mojjominion** [reply]
Yeah that makes sense. Thanks @TkDodo for quick reply. :)
